### PR TITLE
WIP: Get rid of call compile's

### PR DESCRIPTION
**Pull request by dedmen**
The fnc_bi_moduleCurator change might be problematic as parseSimpleArray doesn't allow spaces after comma's. But users might try to use spaces in the textbox in the module.
Same for ace_map_gestures_fnc_moduleGroupSettings

### CODE REVIEW

**Comment by dedmen** [comment]
ace_common_fnc_sendRequest actually didn't conform to the parameters described in the header?

**Comment by commy2** [comment]
Scary PR. Pls don't break anything.

File: addons/common/functions/fnc_debugModule.sqf [hunk 1]
```diff
@@ -17,5 +17,5 @@
 
 params ["_entity"];
 
-GVAR(LOGDISPLAY_LEVEL) = call compile (_entity getVariable ["logDisplayLevel","4"]);
-GVAR(LOGLEVEL) = call compile (_entity getVariable ["logLevel","4"]);
+GVAR(LOGDISPLAY_LEVEL) = (parseSimpleArray format ["[%1]", _entity getVariable ["logDisplayLevel","4"]]) select 0;
+GVAR(LOGLEVEL) = (parseSimpleArray format ["[%1]", _entity getVariable ["logLevel","4"]]) select 0;
```
Comment (commy2) [inline addons/common/functions/fnc_debugModule.sqf]: PSA breaks when there are spaces.

File: addons/mk6mortar/functions/fnc_turretDisplayLoaded.sqf [hunk 2]
```diff
@@ -45,13 +45,7 @@ private _fnc_hideControl = {
...
         _pos = [-9,-9,0,0];
     } else {
-        {
-            if (isNumber (_path >> _x)) then {
-                _pos pushBack (getNumber (_path >> _x));
-            } else {
-                _pos pushBack (call compile (getText (_path >> _x)));
-            };
-        } forEach ["x", "y", "w", "h"];
+        _pos = ["x", "y", "w", "h"] apply {getNumber (_path >> _x)};
```
Comment (commy2) [inline addons/mk6mortar/functions/fnc_turretDisplayLoaded.sqf]: No longer supports globals, as they are read from parsing namespace, unless you change it to the current namespace via call compile.

File: addons/common/functions/fnc_debugModule.sqf (hunk 1 above)
Comment (dedmen) [reply addons/common/functions/fnc_debugModule.sqf]: Yes. But there are none here. Input should be a number. Spaces in there are nonsense.

File: addons/mk6mortar/functions/fnc_turretDisplayLoaded.sqf (hunk 2 above)
Comment (PabstMirror) [reply addons/mk6mortar/functions/fnc_turretDisplayLoaded.sqf]: I have no clue why I wrote it like that, this change should be fine

Comment (dedmen) [reply addons/mk6mortar/functions/fnc_turretDisplayLoaded.sqf]: The config entries are for Arma Engine. And Arma reads it with parsingNamespace too.  And it works for Arma too. So no problem here.

**Comment by dedmen** [comment]
As I wrote in first comment.
We need to check module stuff whether the user can generate input that would fail. Should've marked as WIP.

File: addons/zeus/functions/fnc_bi_moduleCurator.sqf [hunk 3]
```diff
@@ -240,7 +240,7 @@ if (_activated) then {
         _addons = [];
         {
             if (typeof _x == "ModuleCuratorAddAddons_F") then {
-                _paramAddons = call compile ("[" + (_x getvariable ["addons",""]) + "]");
+                _paramAddons = parseSimpleArray ("[" + (_x getvariable ["addons",""]) + "]");
```
Comment (SilentSpike) [inline addons/zeus/functions/fnc_bi_moduleCurator.sqf]: Optimising BI code, I like it 😆

Comment (PabstMirror) [reply addons/zeus/functions/fnc_bi_moduleCurator.sqf]:
```
test getvariable ["addons",""]
"""ace_medical"", ""ace_zeus"""

call compile ("[" + (test getvariable ["addons",""]) + "]");
["ace_medical","ace_zeus"]

parseSimpleArray ("[" + (test getvariable ["addons",""]) + "]");
["ace_medical",<null>]
```
> Error Undefined variable in expression: _x 
 File z\ace\addons\zeus\functions\fnc_bi_moduleCurator.sqf, line 245

Comment (dedmen) [reply addons/zeus/functions/fnc_bi_moduleCurator.sqf]: The whitespace after comma 🤦‍♂️ 
I asked BI whether they'll remove that. See my comment below

**Comment by dedmen** [comment]
This PR has become kinda obsolete now that BI will have fixed the compile memory leak.
All this would do is a performance improvement now, and none of the changes here are on a critical path.

File: addons/zeus/functions/fnc_bi_moduleCurator.sqf (hunk 3 above)
Comment (dedmen) [reply addons/zeus/functions/fnc_bi_moduleCurator.sqf]: I'd say I revert the non 100% sure changes and we only leave the addAction changes from common?

**Comment by TheMagnetar** [comment]
@dedmen should we close this?

**Comment by dedmen** [comment]
No memory leak anymore.
These changes are still performance optimizations though:
https://github.com/acemod/ACE3/pull/6810/files#diff-b406794a3deef0d051b1c023f6f5230aL48
https://github.com/acemod/ACE3/pull/6810/files#diff-6344ea9932824bd79a1f90f9e02b8cd9L57
https://github.com/acemod/ACE3/pull/6810/files#diff-5ea6697c61f916ce450e5117a2f77ea2R26
https://github.com/acemod/ACE3/pull/6810/files#diff-00295e150cdf69d75dfe3d1cc813425cR48
https://github.com/acemod/ACE3/pull/6810/files#diff-1897f2bb4dbe630cd59a18322f8e27c4R55

But not worth the effort for me to update this PR to remove the parseSimpleArray stuff and test the rest.
Saving like a quarter of a millisecond per call on these compiles.

File: addons/mk6mortar/functions/fnc_turretDisplayLoaded.sqf (hunk 2 above)
Comment (commy2) [reply addons/mk6mortar/functions/fnc_turretDisplayLoaded.sqf]: You're right. I was thinking about config entries read by the engine and not getNumber. OOps.

**Pull request closed without merging**